opensearch/src/main/java/org/opensearch/sql/opensearch/executor/OpenSearchExecutionEngine.java

```diff
@@ -49,7 +51,8 @@ public void execute(PhysicalPlan physicalPlan, ExecutionContext context,
               result.add(plan.next());
             }
 
-            QueryResponse response = new QueryResponse(physicalPlan.schema(), result);
+            QueryResponse response = new QueryResponse(physicalPlan.schema(), result,
+                plan.getTotalHits(), planSerializer.convertToCursor(plan));
```

Comment:
what does the totalHits in QueryResponse? does it means result size? if yes, we could get if from result.

Reply:
I'm also confused about why we need to maintain `totalHits` in each physical operator.

> total hits == number of rows matching the search criteria

Do you mean total hits = matched #rows in each page or totally? If the latter, why it becomes 0 finally?

Reply:
total hits = matched #rows totally.
We're following [scroll API](https://www.elastic.co/guide/en/elasticsearch/client/java-api/current/java-search-scrolling.html). OpenSearch returns no hits on the last page - it is a sign that pagination finished. We can store this in a cursor (serialize), but such micro-feature could be implemented as a follow-up work if needed.

Reply:
Does JDBC use totalHits() to determined does the next page exist? Does the response include 0 rows means same thing?

Reply:
JDBC ignores these numbers and checks only row count in each response. This should be fixed in JDBC along with https://github.com/opensearch-project/sql-jdbc/issues/62

Reply:
What if we remove the totoalHits from response? We could take a look other DB, for instance, MySQL.
We could make the change in JDBC and make it match the change in REST api.